.github/workflows/ci-cd.yml

```diff
@@ -1,59 +1,45 @@
-on: [push, pull_request]
-
 name: CI-CD
 
+on:
+  # Triggered on push and pull request events
+  push:
+  pull_request:
+  # Allow manual runs
+  workflow_dispatch:
+  # Monthly runs on the 1st day of the month at midnight
+  schedule:
+    - cron:  '0 0 1 * *'
+
 jobs:
   CI-CD:
     runs-on: ubuntu-latest
-    steps:
 
-      - uses: actions/checkout@v2
+    env:
+      # Access token for GitHub
+      GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
+      # Preserve package sources for informative references in case of errors
+      R_KEEP_PKG_SOURCE: yes
+      # rsconnect is stricter than pak and would complain, see https://github.com/r-lib/pak/issues/485
+      PKG_INCLUDE_LINKINGTO: true
+
+    steps:
 
-      - uses: r-lib/actions/setup-r@master
+      - uses: actions/checkout@v3
 
-      - name: Query dependencies
-        run: |
-          install.packages("remotes")
-          saveRDS(remotes::dev_package_deps(dependencies = TRUE), "depends.Rds", version = 2)
-        shell: Rscript {0}
+      - uses: r-lib/actions/setup-r@v2
+        with:
+          use-public-rspm: true
 
-      - name: Cache R packages
-        uses: actions/cache@v1
+      - uses: r-lib/actions/setup-r-dependencies@v2
         with:
-          path: ${{ env.R_LIBS_USER }}
-          key: ${{ runner.os }}-r-${{ hashFiles('depends.Rds') }}
-          restore-keys: ${{ runner.os }}-r-
+          extra-packages: any::rcmdcheck, any::rsconnect
 
-      - name: Install system dependencies
-        env:
-          RHUB_PLATFORM: linux-x86_64-ubuntu-gcc
-        run: |
-          Rscript -e "remotes::install_github('r-hub/sysreqs')"
-          sysreqs=$(Rscript -e "cat(sysreqs::sysreq_commands('DESCRIPTION'))")
-          sudo -s eval "$sysreqs"
-          # rsconnect dependencies
-          sudo apt-get update && sudo apt-get install -y --no-install-recommends \
-            libxml2-dev libcurl4-openssl-dev
-
-      - name: Install dependencies
-        run: |
-          remotes::install_deps(dependencies = TRUE)
-          remotes::install_cran(c("rcmdcheck", "rsconnect"))
-        shell: Rscript {0}
-
-      - name: Check
-        run: rcmdcheck::rcmdcheck(args = "--no-manual", error_on = "warning", check_dir = "check")
-        shell: Rscript {0}
+      - uses: r-lib/actions/check-r-package@v2
 
       - name: Deploy to shinyapps.io
-        if: github.ref == 'refs/heads/master'
+        if: startsWith(github.ref, 'refs/heads/') || github.ref == 'refs/heads/main' || github.ref == 'refs/heads/master'
```

Comment:
Remove `startsWith(github.ref, 'refs/heads/')` before merging, to revert deploying from the main branch only

Reply:
Done, as well as switching back deployment from rTRhexNG-test to rTRhexNG